src/Chapter4.hs

```diff
@@ -268,6 +282,13 @@ instance Functor Maybe where
 @
 -}
 
+data Maybe' a = Just' a | Nothing' deriving Show
+
+instance Functor Maybe' where
+    fmap :: (a -> b) -> Maybe' a -> Maybe' b
+    fmap f (Just' a) = Just' (f a)
+    fmap _ Nothing' = Nothing'
```

Comment:
Just curious, why did you implement your own version of `Maybe` and an instance for it? For practice purposes? :slightly_smiling_face: